state/indexer/block/kv/kv.go

```diff
@@ -82,8 +87,138 @@ func (idx *BlockerIndexer) Index(bh types.EventDataNewBlockEvents) error {
 	if err := idx.indexEvents(batch, bh.Events, height); err != nil {
 		return fmt.Errorf("failed to index FinalizeBlock events: %w", err)
 	}
+	err = batch.WriteSync()
```

Comment:
You can revert this change because err will always have the value of `batch.WriteSync()` at this point (if it was not nil in line 87, it would have returned.